enumerable.rb

```diff
@@ -0,0 +1,233 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
```

Comment:
* [ ] Improve this by passing the name of the method as an argument to ```to_enum(method_name)```.```return to_enum(method_name) unless block_given?``` instead of ```return to_enum unless block_given?```

Reply:
done